lib/posix/posix.nim

```diff
@@ -985,12 +985,18 @@ proc utimes*(path: cstring, times: ptr array[2, Timeval]): int {.
 proc handle_signal(sig: cint, handler: proc (a: cint) {.noconv.}) {.importc: "signal", header: "<signal.h>".}
 
 template onSignal*(signals: varargs[cint], body: untyped) =
-  ## Setup code to be executed when Unix signals are received. Example:
-  ## from posix import SIGINT, SIGTERM
-  ## onSignal(SIGINT, SIGTERM):
-  ##   echo "bye"
+  ## Setup code to be executed when Unix signals are received. The
+  ## currently handled signal is injected as ``s`` into the calling
+  ## scope.
+  ##
+  ## Example:
+  ##
+  ## .. code-block::
+  ##   from posix import SIGINT, SIGTERM
+  ##   onSignal(SIGINT, SIGTERM):
+  ##     echo "bye from signal ", s
 
-  for s in signals:
+  for s {.inject.} in signals:
     handle_signal(s,
       proc (sig: cint) {.noconv.} =
         body
```

Comment:
This should not allow to access ``s`` within ``body`` (that looks like a compiler bug). ``sig`` needs to be annotated with ``.inject`` here.

Reply:
Oh, I see. Makes sense I guess. I tried annotating `sig` first, but 
```nim
proc (sig {.inject.}: cint) {.noconv.} = 
```
just greeted me with 
```
identifier expected, but found 'sig {.inject.}'
```
Is that not supported for the arguments here? I can workaround with 
```nim
let s {.inject.} = sig
```
in the anonymous proc's body of course.

Reply:
Yeah, too bad, I thought I had implemented that. Anyhow, your workaround is good but the injected name should be ``sig`` IMO.

Reply:
Done.